gis_exercise_2

```diff
@@ -18,7 +18,75 @@
 # https://github.com/AutoGIS-2017/Exercise-2/blob/master/create_polygon.py
 # Each step that you need to fill is marked with capital P -letter (P1 to P6).
 
-# Problem 2: Points to map 
+import os
+os.chdir("C:\\.\\Exercise_2")
```

Comment:
Do not do `chdir` inside your code. Unless you're writing system code, you probably don't need to do this (and you definitely don't need to do it here). Your code should work from the directory you call it in and all your paths should be relative to your project's "root" directory.

What is the purpose of changing the directory here?

Reply:
1. Aw shit I thought I remembered this time! :D so sry! I really need to remember these things.

2. I am using Python notebook for coding and just a normal text editor or csv for the data I import..not sure what you mean with the text editor

3. I think I just wanted to check it out... I thought of it more like the directory in STATA which just specifies where my input/output goes to. So what you say, I should definitely have the .py file at the same location where also all my data / graphs input output goes to? Why is this so mandatory? Either way, then I start doing it!

Reply:
> Aw shit I thought I remembered this time! :D so sry! I really need to remember these things.

haha, no worries, I asked cause there may be a reason I don't know about

> I am using Python notebook for coding and just a normal text editor (notepad) or csv for the data I import..not sure what you mean with the text editor

That makes sense. A notebook is common in scientific programming, so it's good you get familiar with them. However, in "regular" programming you write text files (hence a text editor) and "run" them on the shell. Discussions about which text editor is better are HUGE amongst programmers, haha.

Anyway, if you want to try a text editor to edit your files (so that it highlights the syntax, errors, etc) you can try something simple like Sublime or Atom... Or an IDE, like PyCharm or VSCode. I use PyCharm a lot, and I'd suggest you try it. Just the error highlighting helps a lot to see when something is wrong. 

> So what you say, I should definitely have the .py file at the same location where also all my data / graphs input output goes to? 

Definitely is a strong word. It all depends on what you're trying to do. But I'd say it's a "best practice" to do so.

I'd say a simple solution is to add a directory for the data, and another one for outputs. I did something like that in my pull request: https://github.com/SGJJSR/Python_challenges/pull/8 

> Why is this so mandatory? 

Again, I wouldn't say mandatory, but It's just more organized, not a lot of files everywhere. For example, you can delete everything inside the outputs directory, cause you know that they will be generated by the script. It also helps when other people (me, in this case) is going to look at your code. Think of it this way: if your code was going to be on the internet and you or someone else was going to have to pick up where you left off in a couple of years... what would you like for them to find so that they can start working as soon as possible?

There's a famous saying that says:

> "Always code as if the guy who ends up reading your code will be a violent psychopath who knows where you live."